src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -2802,22 +2820,31 @@ module Queries = struct
       ~resolve:(fun {ctx= coda; _} () max_length ->
         let open Option.Let_syntax in
         let%map best_chain = Mina_lib.best_chain ?max_length coda in
-        List.map best_chain ~f:(fun breadcrumb ->
-            let hash = Transition_frontier.Breadcrumb.state_hash breadcrumb in
-            let transition =
-              Transition_frontier.Breadcrumb.validated_transition breadcrumb
-            in
-            let transactions =
-              Mina_transition.External_transition.Validated.transactions
-                ~constraint_constants:
-                  (Mina_lib.config coda).precomputed_values
-                    .constraint_constants transition
-            in
-            With_hash.Stable.Latest.
-              { data=
-                  Filtered_external_transition.of_transition transition `All
-                    transactions
-              ; hash } ) )
+        List.map best_chain ~f:(block_of_breadcrumb coda) )
+
+  let block =
+    field "block"
```

Comment:
Since there are multiple places where a failure can be introduced in the processing of this query -- can you make this a result_field instead and provide proper error messages for each of the places where we (currently) are binding over the option? Then a consumer can understand why it's failing.

Reply:
@bkase Do you mean a distinct query, like `blockByHeight`, or that height should be an additional optional argument?